frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -228,6 +231,48 @@ const defaultSamples = ImmutableMap<GroupVersionKind, Sample[]>()
         targetResource: getTargetResource(ConsoleLinkModel),
       },
     ],
+  )
+  .setIn(
+    [referenceForModel(ConsoleModel)],
+    [
+      {
+        title: 'Add catalog categories',
+        description:
+          'Provides a list of default categories which are shown in the Developer Catalog.',
+        id: 'default',
+        snippet: true,
+        lazyYaml: () => {
+          const mapSubcategory = (subcategory) => ({
+            id: subcategory.id,
+            label: subcategory.label,
+            tags: !subcategory.field || subcategory.field === 'tag' ? subcategory.values : [],
+          });
+
+          const mapCategory = (category) => ({
+            id: category.id,
+            label: category.label,
+            tags: !category.field || category.field === 'tag' ? category.values : [],
+            subcategories: category.subcategories
+              ? Object.values(category.subcategories).map(mapSubcategory)
+              : [],
+          });
+
+          const categories = Object.values(catalogCategories).map(mapCategory);
```

Comment:
Why bother with this conversion. Shouldn't the mapping be 1:1 between the default data structure and the yaml?

Reply:
Rebased this PR now on #7196 where I changed all `defaultCatalogCategories` to the same structure which is used by the CRD. After this I now removed all the mapping from this PR and just return this YAML snippet:

```tsx
          const yaml = {
            spec: {
              customization: {
                developerCatalog: {
                  categories: defaultCatalogCategories,
                },
              },
            },
          };
          return YAML.dump(yaml);
```

Reply:
Changed this to 

```tsx
return YAML.dump(defaultCatalogCategories);
```

Because the editor keeps indentation fine.